cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -142,10 +142,13 @@ func (k *proxy) GetConfig() (*rest.Config, error) {
 	}
 	restConfig, err := clientcmd.NewDefaultClientConfig(*config, configOverrides).ClientConfig()
 	if err != nil {
-		if strings.HasPrefix(err.Error(), "invalid configuration:") {
-			return nil, errors.New(strings.Replace(err.Error(), "invalid configuration:", "invalid kubeconfig file; clusterctl requires a valid kubeconfig file to connect to the management cluster:", 1))
+		if !strings.HasPrefix(err.Error(), "invalid configuration:") {
+			return nil, err
+		}
+		restConfig, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, err
```

Comment:
Let's return an error that expresses that both the kubeconfig and the in cluster config didn't work (previosu error from l.146 + something for in cluster

Reply:
Updated the error message to "invalid kubeconfig file and failed to create inClusterConfig as well"